### PR TITLE
Add test for observation repository

### PR DESCRIPTION
**Pull request by garciampred**

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf-projects/cads-api-client?pullRequest=50) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf-projects/cads-api-client?pullRequest=50) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf-projects/cads-api-client?pullRequest=50) it.</sub>

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -174,7 +174,7 @@ jobs:
         python -m pip install --no-deps -e .
     - name: Run tests
       run: |
-        make unit-tests COV_REPORT=xml
+        make integration-tests COV_REPORT=xml
```
Comment (malmans2) [inline .github/workflows/on-push.yml:177]:
```suggestion
        make unit-tests COV_REPORT=xml
```
Hi @garciampred,

Please revert this change. In general, we control all configuration files with Cookiecutter + Cruft.

`make integration-tests` is not meant to be executed from GitHub CI. We only use it to test development branches and to ensure everything is in good shape before a release. For example, you can run it out-of-the-box from dev-explorer or locally with an appropriate `.cads-api-client.json`.

We use the integration tests on GitHub to test different Python versions and/or environments with different dependencies for packages that have optional dependencies.

**Review by malmans2 — CHANGES_REQUESTED** [review]

File: .github/workflows/on-push.yml (hunk 1 above)
Comment (garciampred) [reply .github/workflows/on-push.yml:177]: ok sorry, I was just testing, the PR is not finished

Comment (malmans2) [reply .github/workflows/on-push.yml]: OK, sorry about that.
This is a public repo:  You should be able to use Draft PR when you are just tinkering.

**Review by malmans2 — APPROVED** [review]
Thanks, I isolated the obs tests in a separate file.
Merging!

**Pull request merged**